### PR TITLE
feat:check registry login status

### PR DESCRIPTION
**Pull request by kakaZhou719**
### Describe what this PR does / why we need it
feature:
1, add prechecker 
2，check registry auth info before run cloud build

### CODE REVIEW

File: check/checker/register_checker.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+package checker
+
+import (
+	"fmt"
+
+	"github.com/alibaba/sealer/common"
+	"github.com/alibaba/sealer/utils"
+)
+
+type RegisterChecker struct {
```
Comment (justadogistaken) [inline check/checker/register_checker.go:24]: Registry?

File: check/checker/register_checker.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+func (r *RegisterChecker) Check() error {
+	// check register info;
+	authFile := common.DefaultRegistryAuthConfigDir()
+	if !utils.IsFileExist(authFile) {
+		return fmt.Errorf("register auth info not found,please run 'sealer login' first")
+	}
```
Comment (justadogistaken) [inline check/checker/register_checker.go:32]: Even though the auth info exists, there still be some possibilities that causing auth failure.

File: check/checker/register_checker.go (hunk 1 above)
Comment (kakaZhou719) [reply check/checker/register_checker.go]: fixed

File: check/checker/registry_checker.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+type RegistryChecker struct {
+}
+
+func (r *RegistryChecker) Check() error {
+	// check registry info;
+	authFile := common.DefaultRegistryAuthConfigDir()
+	if !utils.IsFileExist(authFile) {
+		return fmt.Errorf("registry auth info not found,please run 'sealer login' first")
+	}
```
Comment (justadogistaken) [inline check/checker/registry_checker.go:32]: still need to login to check the auth info?

File: check/checker/types.go [hunk 4]
```diff
@@ -16,6 +16,12 @@ package checker
 
 import v1 "github.com/alibaba/sealer/types/api/v1"
 
+// Checker: checking cluster status,such as node,pod,svc status.
 type Checker interface {
 	Check(cluster *v1.Cluster) error
 }
+
+// PreChecker: checking prerequisite for sealer,such as system info,register login status.
+type PreChecker interface {
```
Comment (fanux) [inline check/checker/types.go:25]: Why don't using the Checker interface? Some prechecker needs cluster.

Comment (kakaZhou719) [reply check/checker/types.go:25]: i think prechecker is a static checker ,such as systeminfo . but for cluster its a running state parameter。

File: check/checker/registry_checker.go (hunk 3 above)
Comment (kakaZhou719) [reply check/checker/registry_checker.go:32]: for now ,only check the existence of docker.json ,as issue #108,this PR is not a proper way to do it. i will refactor it.

File: check/checker/types.go (hunk 4 above)
Comment (fanux) [reply check/checker/types.go:25]: Some checker can ignore this argument. All the checkers implement the same interface

**Review by fanux — APPROVED** [review]

File: check/checker/registry_checker.go (hunk 3 above)
Comment (fanux) [reply check/checker/registry_checker.go:32]: I think need to log in to check it.

**Review by justadogistaken — APPROVED** [review]

**Pull request merged**